drafts/AENS.md

```diff
@@ -0,0 +1,414 @@
+transition into the `revoked` state. Once it reaches this state, the name 
+will be released, i.e. transition into the `unclaimed` state, after which 
+it could be claimed again.
+
+To prevent the entry from expiring a user can, at any point before reaching
+the expiration date, update the entry, which pushes the expiration date
+into the future by the time delta of the registration time and the update
+time, e.g. if a user posts an update one week after claiming their name,
+the expiration date gets pushed one week further into the future.
+The main motivation of this expiration date is to prevent situations where
+the private keys, which control the name, are lost making the name unusable 
+as well.
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", 
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be 
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+
+### AENS Entry
+
+```
+  name         size (bytes)
+ ------------ ----
+| owner      | 32 |
+ ------------ ----
+| expires_by | 32 |
+ ------------ ----
+| ttl        | 8  |
+ ------------ ----
+| pointers   |    |
+ ------------ ----
+```
+
+***owner***: the account, which controls this entry.
+
+***expires_by***: the blockheight after which the entry goes
+into the `revoked` state.
+
+***ttl***: a suggestion as to how long any clients should 
+cache this information. (***TODO***: should have a reasonable 
+upper limit, e.g. 86400 seconds)
+
+***pointers***: a tuple with all the values this entry
+points towards, e.g. `{kind: "ipfs", data: "QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC"}`
+
+
+### Name
+
+Names MUST be normalized according to IDNA2008 before hashing, as 
+described in [uts-46](https://unicode.org/reports/tr46) and SHOULD 
+be stored in their ASCII representation.
+
+See also [rfc3491](https://tools.ietf.org/html/rfc3491) and 
+[rfc3492](https://tools.ietf.org/html/rfc3492).
+
+Names SHOULD be at most 253 characters long, in order to enable 
+interoperability with DNS.
+
+***Note:*** Fot this first draft names MUST have `.aet` or `.test`
+as a suffix, i.e. `mywallet.aet`.
+
+
+### Namespaces/Labels
+
+A name is split up into labels by the `FULL STOP (U+002E .)` character.
+These labels will also be referred to as namespaces, e.g. `mywallet.aet` 
+can be understood as the `mywallet` namespace in the `aet` namespace.
+
+A label MUST be longer than three characters.
+
+Labels SHOULD be at most 63 characters long and the full path
+of all names and namespaces SHOULD be at most 253 characters
+long, in order to enable interoperability with DNS.
+
+
+### Registrars
+
+A registrar controls who is allowed to claim names in their namespace
+and under what circumstances they are allowed to do so.
+
+The only available registrars for this AENS version will be hard-coded
+ones, which own the `.`, `.aet` and `.test` namespaces.
```

Comment:
why do we want to have a `.test` namespace in the mainnet? I'd prefer having `.aet` in the mainnet and `.test` in the testnet

Reply:
why we need `.test` for testnet? the `.aet` in testnet will be really test-`.aet`?

Reply:
Since there will be a time (nlockheight) sometime in the future when .aet will be available we can have exactly the same code for the test net and the main net. To test nsmes before that height both domains are needed.